crates/maybenot/src/time.rs

```diff
@@ -0,0 +1,61 @@
+use std::ops::AddAssign;
+
+/// Trait representing instants in time. Allows using maybenot frameworks with
+/// custom time sources. If you want to use maybenot with a different time source
+/// than `std::time::Instant`, implement this trait for your instant type, and the
+/// [`Duration`] trait for your corresponding duration type.
+pub trait Instant: Clone + Copy {
+    type Duration: Duration;
+
+    /// Returns the amount of time elapsed from another instant to this one.
+    ///
+    /// Should return a zero duration if `earlier` is later than `self`
```

Comment:
@pylls Do you have an opinion on the desired behavior of this? I documented and implemented it as how the current behavior already was. Rust has returned zero durations if `earlier` is later than `self` since Rust 1.59 (very old). Before that it panicked instead. I personally feel it's saner to define it in this stricter way.

Reply:
My two cents: It's fine to saturate, and it's fine to return an `Option<Duration>`, but panicking behavior is undesirable.

IMO: the method name should probably correspond to whichever corresponding `std::time::Instant` method has the same behavior.  So if it's doing to saturate, it should be called `saturating_duration_since`, and if it's going to return an `Option` this method should be called `checked_duration_since`.

Reply:
I think we should use saturating unless there is a need why we would need to panic when a user sends in instants going backwards. This currently does use the same method name as `std`. But one can of course argue that that name is what it is for historical reasons and that `saturating_duration_since` is more clear. But neither `std` nor `coarsetime` calls it that really :)

Reply:
(Hi! Thanks for all of our efforts @faern and @nmathewson , will get to all of this tonight.)

Saturation, for sure, is the way to go.  Panics are terrible. Internally the framework already clamps all distributions to a minimum of 0.0.  The simulator is also dependent on the current std behaviour.

Reply:
I renamed the method to `saturating_duration_since` and changed the impl to use that method directly from `std::time::Instant`. I forgot that exists :)